### PR TITLE
Updated section about changing small files

### PR DESCRIPTION
**Pull request by nollety**
# Description

This is a suggestion to update the developer guide - data guide section about changing small files.
I was unsure about the exact steps to follow when updating small data files, I thought we could give a little more details.

# Checklist

- [x] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [x] The code is appropriately documented
- [x] The code is tested to prove its function
- [x] The feature branch is rebased on the current state of the `main` branch
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

### CODE REVIEW

File: docs/rst/developer_guide/data_guide.rst [hunk 1]
```diff
@@ -83,10 +83,65 @@ Modifying the data
...
 ``small_files``
-    Install `pre-commit <https://pre-commit.com/>`_ and activate the hooks, this
-    will update the registry automatically upon commit. The rules used to create
-    the registry file are defined in the ``"registry_rules.yml"`` file. If, for
-    some reason, you cannot use pre-commit, then you must be very careful and
+    Install `pre-commit <https://pre-commit.com/>`_:
+    
+    .. code:: bash
+
+       pip install pre-commit
```
Comment (leroyvn) [inline docs/rst/developer_guide/data_guide.rst:90]: These installation instructions are just a subset of what the pre-commit website suggests/recommends. I'd rather have here a link to the pre-commit website (telling people how to install third party software isn't really our job IMO).

File: docs/rst/developer_guide/data_guide.rst [hunk 2]
```diff
@@ -83,10 +83,65 @@ Modifying the data
...
+
+       Creating registry file from '.'
+       Using rules in 'registry_rules.yml'
+       Writing registry file to 'registry.txt'
+       100% 181/181 [00:00<00:00, 100859.44it/s]
+
+    The hook script failed because we changed the data and the changes were not commited.
+    This is the expected behaviour.
+    The hook script updated the registry file with the sha256 sum of the data file we added.
+    Now add the changes to the registry file and commit:
```
Comment (leroyvn) [inline docs/rst/developer_guide/data_guide.rst:123]:
```suggestion
    Now add the changes to the registry file and commit again:
```

File: docs/rst/developer_guide/data_guide.rst [hunk 3]
```diff
@@ -83,10 +83,65 @@ Modifying the data
...
+
+       Update registry..........................................................Passed
+       [master 0b9c760] Added some data
+       2 files changed, 2 insertions(+)
+       create mode 100644 spectra/some_data.nc
+
+    The rules used to create the registry file are defined in the
+    ``"registry_rules.yml"`` file.
+    Be aware that if you add a data file that is not included by these rules, it will
+    not be registered therefore it will not be accessible by the data store.
```
Comment (leroyvn) [inline docs/rst/developer_guide/data_guide.rst:142]:
```suggestion
    not be registered and therefore it will not be accessible by the data store.
```

**Review by leroyvn — COMMENTED** [review]
Hi @nollety, this will indeed be useful for pre-commit newbies. I made a couple of comments, we can merge this when they will be addressed.

File: docs/rst/developer_guide/data_guide.rst (hunk 1 above)
Comment (nollety) [reply docs/rst/developer_guide/data_guide.rst:90]: That's right, I agree

**Comment by nollety** [comment]
@leroyvn thanks for your feedback, I applied your suggestions and since it's only a doc update, I took the liberty to merge

**Pull request merged**